### ISSUE TITLE
NNBD. No type promotion for null-aware cascade operator 

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#39141]
```dart
class C {
  String test = "Show must go on";
}

main() {
  C? c = null;
  c ?.. test;
}
```
Analyzer with --enable-experiment=non-nullable option produces "  error - The expression is nullable and must be null-checked before it can be used.". There must be no error here,

dartanalyzer version 2.6.0-dev.8.2

**Comment by lrhn** [comment]
I hit this too while migrating platform libraries. The workaround was easy: if (c != null) c..test...`, so it wasn't blocking me, but there was code hitting the issue.

**Comment by scheglov** [comment]
The example works for me - no analysis errors.
I believe @stereotype441 fixed this in 7d4bc6a1df70173226ba996db7962a45e0beea77.
